spec.md

```diff
@@ -1925,6 +1929,123 @@ To disallow infinitely recursive hardware, modules cannot contain instances of
 itself, either directly, or indirectly through instances of other modules it
 instantiates.
 
+## Optional Groups
```

Comment:
should this Optional Group chapter go before ExtModule since we refer to it in the ExtModule chapter?

Reply:
I'll rework this to do that. This section was written before the `declgroup` declaration was added which resulted in the ordering.